### PR TITLE
Remove ephemeral peers with closed outbound connections

### PR DESCRIPTION
**Pull request by prashantv**

### CODE REVIEW

File: peer.go [hunk 1]
```diff
@@ -399,41 +397,43 @@ func (p *Peer) AddOutboundConnection(c *Connection) error {
...
 }
 
-// checkInboundConnection will check whether the changed connection is an inbound
-// connection, and will remove any closed connections.
-func (p *Peer) checkInboundConnection(changed *Connection) (updated bool, isInbound bool) {
-	newConns := p.inboundConnections[:0]
-	for _, c := range p.inboundConnections {
+// removeClosed will check remove the changed connection if it exists
+// in the specified connections list.
+func (p *Peer) removeClosed(connsPtr *[]*Connection, changed *Connection) (updated bool, found bool) {
```
Comment (akshayjshah) [inline peer.go]: Is there any value in protecting ourselves from a nil `connsPtr`? Seems like

```
if connsPtr == nil {
  return false, false
}
```

is easy enough.

File: peer.go [hunk 2]
```diff
@@ -399,41 +397,43 @@ func (p *Peer) AddOutboundConnection(c *Connection) error {
...
-		}
+		updated = true
 	}
 	if updated {
-		p.inboundConnections = newConns
+		*connsPtr = newConns
 	}
 
-	return updated, isInbound
+	return updated, found
```
Comment (akshayjshah) [inline peer.go]: I think I'm missing something here - don't we need to append the non-matching conns to `newConns` somewhere?

Comment (prashantv) [inline peer.go]: I have no idea how the tests are even passing

File: peer.go (hunk 1 above)
Comment (prashantv) [inline peer.go]: Not too worried about this, if it's nil, I'd be fine with crashing, since it's a pretty bad code bug.

Comment (akshayjshah) [inline peer.go]: Also, would this be better named `removeConnection`? I don't see any reason that `changed` needs to be a closed connection.

Comment (prashantv) [inline peer.go]: yep, that's true. renaming to `removeConnection`

**Comment by prashantv** [comment]
Fixed and renamed `removeConnection`, totally rewrote it actually to be simpler. Updated now, PTAL

**Comment by akshayjshah** [comment]
:shipit:

**Pull request merged**